delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViewReferenceExchange.scala

```diff
@@ -0,0 +1,65 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph
+
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model.{BlazegraphView, BlazegraphViewRejection, ViewResource}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.sdk.ReferenceExchange
+import ch.epfl.bluebrain.nexus.delta.sdk.ReferenceExchange.ReferenceExchangeValue
+import ch.epfl.bluebrain.nexus.delta.sdk.implicits._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment.IriSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, ResourceRef}
+import monix.bio.{IO, UIO}
+
+/**
+  * BlazegraphView specific [[ReferenceExchange]] implementation.
+  *
+  * @param views the blazegraph module
+  */
+class BlazegraphViewReferenceExchange(views: BlazegraphViews)(implicit
```

Comment:
Please find a way to reconcile parts of `EventExchange` with `ReferenceExchange`. 
At least reuse `ReferenceExchangeValue`, since it is exactly what is already being done https://github.com/BlueBrain/nexus/blob/master/delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViews.scala#L314

I think you can replace `EventExchange.StateExchange` for your `ReferenceExchangeValue` and some minor adaptation. This should reduce the code.

Also we either create the `EventExchange` and `ReferenceExchange` implementations always inside the companion object of `Schemas`, `Resolvers`, etc... or as a separate class like here.